Source/RapyutaSimulationPlugins/Public/Robots/RobotEmptyVehicle.h

```diff
@@ -0,0 +1,149 @@
+/**
+ * @file RobotVehicle.h
+ * @brief Base RobotVehicle class. Other robot class should inherit from this class. Example is #ATurtlebotBurger.
+ * @copyright Copyright 2020-2022 Rapyuta Robotics Co., Ltd.
+ */
+
+#pragma once
+
+// UE
+#include "Components/SkeletalMeshComponent.h"
+#include "CoreMinimal.h"
+#include "Engine/TargetPoint.h"
+#include "GameFramework/Pawn.h"
+
+// RapyutaSimulationPlugins
+#include "Core/RRBaseActor.h"
+#include "Drives/RRJointComponent.h"
+#include "Drives/RobotVehicleMovementComponent.h"
+#include "Sensors/RRROS2BaseSensorComponent.h"
+
+// rclUE
+#include "ROS2Node.h"
+
+#include "RobotEmptyVehicle.generated.h"
+
+class URobotVehicleMovementComponent;
+
+/**
+ * @brief Base RobotEmptyVehicle class. Other robot class should inherit from this class.
+ * This actor moves with #URobotVehicleMovementComponent.
+ * This actor is possessed by #ARRRobotVehicleROSController to be control from ROS2.
+ * You can find example at #ATurtlebotBurger.
+ *
+ */
+UCLASS()
+class RAPYUTASIMULATIONPLUGINS_API ARobotEmptyVehicle : public ARRBaseActor

```

Comment:
Since this base vehicle class is not really empty as housing links/joints and a default scene component as root, I'd suggest its name be `ARRRobotBaseVehicle` to make it consistent with the naming convention as in `ARRBaseActor`.

(Regarding `ARobotVehicle, `ASimulationState`, which should be prefixed `RR` as well, however because of the complicated overheads required for child BP class changes, it is not very straightforward)

Reply:
Since this base vehicle class is not really empty as housing links/joints and a default scene component as root, I'd suggest its name be `ARRRobotBaseVehicle` to make it consistent with the naming convention as in `ARRBaseActor`.

(Regarding `ARobotVehicle, ASimulationState`, which should be prefixed `RR` as well, however because of the complicated overheads required for child BP class changes, it is not very straightforward)